mnefun/_mnefun.py

```diff
@@ -414,9 +416,22 @@ def convert_subjects(self, subj_template, struc_template=None):
             Structural template to use.
         """
         if struc_template is not None:
-            self.structurals = [struc_template % subj
-                                for subj in self.subjects]
-        self.subjects = [subj_template % subj for subj in self.subjects]
+            if isinstance(struc_template, string_types):
+                def fun(x):
+                    return struc_template % x
+            else:
+                fun = struc_template
+            new = [fun(subj) for subj in self.subjects]
+            assert all(isinstance(subj, string_types) for subj in new)
+            self.structurals = new
+        if isinstance(subj_template, string_types):
+            def fun(x):
+                return subj_template % x
+        else:
+            fun = subj_template
+        new = [fun(subj) for subj in self.subjects]
+        assert all(isinstance(subj, string_types) for subj in new)
+        self.subjects = new
 
```

Comment:
It allows more flexible mapping between `subj` and `struct` names, e.g. if you need to put in a `subj.upper()` for example